src/entropies_estimators/nearest_neighbors/Goria.jl

```diff
@@ -0,0 +1,66 @@
+using StateSpaceSets: AbstractDataset, Dataset
+using Neighborhood: KDTree, NeighborNumber, Theiler
+using Neighborhood: bulksearch
+using SpecialFunctions: digamma
+
+export Goria
+
+"""
+    Goria <: DifferentialEntropyEstimator
+    Goria(k = 1, w = 0)
+
+The `Goria` estimator computes the [`Shannon`](@ref) differential
+[`entropy`](@ref) of `x` (a multi-dimensional [`Dataset`](@ref)).
+
+## Description
+
+Assume we have samples ``\\{\\bf{x}_1, \\bf{x}_2, \\ldots, \\bf{x}_N \\}`` from a
+continuous random variable ``X \\in \\mathbb{R}^d`` with support ``\\mathcal{X}`` and
+density function``f : \\mathbb{R}^d \\to \\mathbb{R}``. `Goria` estimates
+the [Shannon](@ref) differential entropy
+
+```math
+H(X) = \\int_{\\mathcal{X}} f(x) \\log f(x) dx = \\mathbb{E}[-\\log(f(X))].
+```
```

Comment:
every differential estimator repeat these lines and this formula. Wouldn't it make sense to just put this formula in the diff entropy doc section and not repeat it in each doxcstruing? Like at the start of the doc section you can say: unless noted differently in the estimator docstring, the estimators compute the differential Shannon...

Reply:
I think it's nice if every estimator has its own self-contained description, where the formula it computes is included. Especially when adding estimators for `Tsallis`, `Renyi` and other entropies. This makes it much easier to follow the docstring, at least for some estimators, so you don't have to navigate back and forth. For example, the docs for the order statistics based estimators reads very intuitively.

Can we perhaps re-assess this stylistic choice once we've added all the estimators I've got lined up? It might make more/less sense to shorten them after that.

Reply:
Anyways, since this is a style choice that doesn't affect functionality that this PR introduces, I'll open a general issue for it, and just merge this.